src/twisted/python/systemd.py

```diff
@@ -68,26 +70,70 @@ def fromEnvironment(
         if start is None:
             start = cls._START
 
-        descriptors: List[int] = []
-
-        try:
-            pid = int(environ["LISTEN_PID"])
-        except (KeyError, ValueError):
-            pass
+        if str(getpid()) == environ.get("LISTEN_PID"):
+            descriptors: List[int] = _parseDescriptors(start, environ)
+            names: Sequence[str] = _parseNames(environ)
         else:
-            if pid == getpid():
-                try:
-                    count = int(environ["LISTEN_FDS"])
-                except (KeyError, ValueError):
-                    pass
-                else:
-                    descriptors = list(range(start, start + count))
-                    del environ["LISTEN_PID"], environ["LISTEN_FDS"]
-
-        return cls(descriptors)
-
-    def inheritedDescriptors(self) -> Iterable[int]:
+            descriptors = []
+            names = ()
+
+        # They may both be missing (consistent with not running under systemd
+        # at all) or they may both be present (consistent with running under
+        # systemd 227 or newer).  It is not allowed for only one to be present
+        # or for the values to disagree with each other.
```

Comment:
If no names are defined for a socket, I guess that systemd will just use a generic `1.socket` name.

It would be nice to have a link with this specification, but is not required.